backend/src/test/java/com/onwelo/practice/bts/service/BankAccountServiceTest.java

```diff
@@ -1,24 +1,141 @@
+        ArrayList<Transfer> out = new ArrayList<>(
+                bankAccountService.getBankAccountById(bankIn.getId()).getOutgoingTransfers());
+        Assert.assertNotNull(out);
+        Assert.assertEquals(out.get(0).getId(), transferService.getTransferById(transfer.getId()).getId());
+
+        transferService.deleteTransfer(transfer.getId());
+        bankAccountService.deleteBankAccount(bankIn.getId());
+        bankAccountService.deleteBankAccount(bankOut.getId());
+    }
+
+    @Test
+    public void getBankAccountById() {
+        BankAccount bankAccount = new BankAccount("140159260076545510730339",
                 "Jan", "Kowalski", 1000.0f, 0.0f);
 
         bankAccountService.addBankAccount(bankAccount);
-        Assert.notNull(bankAccountService.getBankAccountById(bankAccount.getId()));
+        BankAccount bankAccount2 = bankAccountService.getBankAccountById(bankAccount.getId());
+        Assert.assertEquals(bankAccount.getId(), bankAccount2.getId());
+
+        bankAccountService.deleteBankAccount(bankAccount.getId());
     }
 
+    @Test
+    public void getBankAccountByNumber() {
+        BankAccount bankAccount = new BankAccount("140159260076545510730339",
+                "Jan", "Kowalski", 1000.0f, 0.0f);
+
+        bankAccountService.addBankAccount(bankAccount);
+        Assert.assertEquals(bankAccount.getId(),
+                bankAccountService.getBankAccountByNumber(bankAccount.getBankNumber()).getId());
+
+        bankAccountService.deleteBankAccount(bankAccount.getId());
+    }
+
+    @Test
+    public void addBankAccount() {
+        BankAccount bankAccount = new BankAccount("140159260076545510730339",
+                "Jan", "Kowalski", 1000.0f, 0.0f);
+
+        bankAccountService.addBankAccount(bankAccount);
+        Assert.assertNotNull(bankAccountService.getBankAccountById(bankAccount.getId()));
+
+        bankAccountService.deleteBankAccount(bankAccount.getId());
+    }
+
+    @Test
+    public void updateBankAccount() {
+        BankAccount bankAccount = new BankAccount("140159260076545510730339",
+                "Jan", "Kowalski", 1000.0f, 0.0f);
+
+        bankAccountService.addBankAccount(bankAccount);
+        bankAccount.setFirstName("Adam");
+        bankAccountService.updateBankAccount(bankAccount);
+        Assert.assertEquals(bankAccount.getFirstName(), bankAccountService.getBankAccountById(bankAccount.getId()).getFirstName());
+
+        bankAccountService.deleteBankAccount(bankAccount.getId());
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void deleteBankAccount() {
+        BankAccount bankAccount = new BankAccount("140159260076545510730339",
+                "Jan", "Kowalski", 1000.0f, 0.0f);
+
+        bankAccountService.addBankAccount(bankAccount);
+        bankAccountService.deleteBankAccount(bankAccount.getId());
+        Assert.assertNull(bankAccountService.getBankAccountById(bankAccount.getId()));
```

Comment:
Nie sprawdziliśmy wcześniej czy konto się zapisało. Jeśli więc zapis nie działa to test pewnie i tak przejdzie.